src/common/meson.build

```diff
@@ -0,0 +1,400 @@
+	return 0;
+}
+'''
+            res = cc.links(bswap_src.format(f), name: f)
+            if res == true
+                config_h.set10('HAVE_' + f.to_upper(), true,
+                    description: 'Define to 1 if ' + f + '() is available.'
+                )
+            endif
+        endforeach
+    endif
+endif
+
+fast_unaligned_access = false
+if get_option('unaligned-access').auto() == true
+    unaligned_access_cpus = [
+        'ppc',
+        'ppc64',
+        'x86',
+        'x86_64',
+    ]
+    foreach cpu : unaligned_access_cpus
+        if host_machine.cpu_family() == cpu
+            config_h.set10('TUKLIB_FAST_UNALIGNED_ACCESS', true,
+                description: 'Define to 1 if the system supports fast unaligned access to 16-bit, 32-bit, and 64-bit integers.'
+            )
+            fast_unaligned_access = true
+            break
+        endif
+    endforeach
+
+    if fast_unaligned_access == false
+        unaligned_access_cpus = [
+            'aarch64',
+            'arm',
+            'riscv32',
+            'riscv64',
+        ]
+        foreach cpu : unaligned_access_cpus
+            if host_machine.cpu_family() == cpu
+                ua_src = '''
+#if !defined(__ARM_FEATURE_UNALIGNED) \
+		&& !defined(__riscv_misaligned_fast) \
+		&& !defined(_MSC_VER)
+compile error
+#endif
+int main(void) { return 0; }
+'''
+                res = cc.compiles(ua_src, name: 'fast unaligned access for ' + cpu)
+                if res == true
+                    config_h.set10('TUKLIB_FAST_UNALIGNED_ACCESS', true,
+                    description: 'Define to 1 if the system supports fast unaligned access to 16-bit, 32-bit, and 64-bit integers.'
+                    )
+                    fast_unaligned_access = true
+                    break
+                endif
+            endif
+        endforeach
+    endif
+elif get_option('unaligned-access').enabled() == true
+    config_h.set10('TUKLIB_FAST_UNALIGNED_ACCESS', true,
+        description: 'Define to 1 if the system supports fast unaligned access to 16-bit, 32-bit, and 64-bit integers.'
+    )
+    fast_unaligned_access = true
+endif
+
+if get_option('unsafe-type-punning') == true
+    config_h.set10('HAVE___BUILTIN_ASSUME_ALIGNED', true,
+        description: 'Define to 1 if the GNU C extension __builtin_assume_aligned is supported.'
+                    )
+endif
+
+# tuklib_mbstr.m4
+
+### FIXME : mbrtowc is supported in C99 spec and on Windows
+if cc.has_header('wchar.h')
+    config_h.set10('HAVE_WCHAR_H', true,
+        description: 'Define to 1 if the header file <wchar.h> is available.'
+    )
+
+    res = cc.has_type('mbstate_t',
+        prefix : '#include <wchar.h>'
+    )
+
+    if res == true
+        if cc.has_function('mbrtowc',
+            prefix : '#include <wchar.h>'
+            )
+            config_h.set10('HAVE_MBRTOWC', true,
+                description: 'Define to 1 if the mbrtowc() function is available.'
+            )
+        endif
+    endif
```

Comment:
```suggestion
    if res and cc.has_function('mbrtowc', prefix : '#include <wchar.h>')
        config_h.set10('HAVE_MBRTOWC', true,
            description: 'Define to 1 if the mbrtowc() function is available.'
        )
    endif
```

Reply:
done in 8f0731e245f0af8fd023877f774006d4346cc2fa